crates/iceberg-catalog/migrations/20250508101407_tasks_store_their_state.sql

```diff
@@ -0,0 +1,134 @@
+create type entity_type as enum ('tabular');
+create type task_final_status as enum ('failed', 'cancelled', 'completed');
+
+alter table task
+    rename column suspend_until to scheduled_for;
+
+alter table task
+    add column task_data         jsonb,
+    add column entity_type       entity_type,
+    add column entity_id         uuid,
+    add column last_heartbeat_at timestamptz,
+    -- we're moving old tasks into task_log and delete entries from task
+    drop constraint task_parent_task_id_fkey,
+    alter column scheduled_for set default now(),
+    alter column scheduled_for set not null;
+
+update task
+set task_data   = jsonb_build_object(
+        'tabular_id', te.tabular_id,
+        'typ', te.typ,
+        'deletion_kind', te.deletion_kind),
+    entity_type = 'tabular',
+    entity_id   = te.tabular_id
+from tabular_expirations te
+where te.task_id = task.task_id;
+
+update task
+set task_data   = jsonb_build_object(
+        'tabular_id', tp.tabular_id,
+        'typ', tp.typ,
+        'tabular_location', tp.tabular_location),
+    entity_type = 'tabular',
+    entity_id   = tp.tabular_id
+from tabular_purges tp
+where tp.task_id = task.task_id;
+
+drop table tabular_expirations;
+drop table tabular_purges;
+
+alter table task
+    alter column task_data set not null,
+    alter column entity_type set not null,
+    alter column entity_id set not null;
+
+create table task_config
+(
+    warehouse_id uuid references warehouse (warehouse_id) on delete cascade not null,
+    queue_name   text                                                       not null,
+    config       jsonb                                                      not null,
+    max_age      interval,
```

Comment:
What do you think about `task_instance_heartbeat_timeout` instead of `max_age` now that we have repurposed it?
Airflow has many different timeouts, the most important ones are:
1. [`task_instance_heartbeat_timeout`](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#task-instance-heartbeat-timeout) (Timeout since Heartbeat)
1. [`execution_timeout`](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/tasks.html#timeouts) (Timeout since pickup)
1. [`timeout`](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/tasks.html#timeouts) Timeout since task creation.

I would assiciate `max_age` more with 2) or 3) than with 1). I think in Lakekeeper we need 1) and 2).

Reply:
Specifying 2) is much more important for users than 1) typically.